api/admin/controller.py

```diff
@@ -1906,4 +1908,34 @@ def discovery_services(self):
         else:
             return Response(unicode(_("Success")), 200)
 
+    def library_registrations(self, do_get=HTTP.get_with_timeout, do_post=HTTP.post_with_timeout):
+        if flask.request.method == "POST":
+
+            integration_id = flask.request.form.get("integration_id")
+            library_short_name = flask.request.form.get("library_short_name")
+
+            integration = get_one(self._db, ExternalIntegration,
+                                  goal=ExternalIntegration.DISCOVERY_GOAL,
+                                  id=integration_id)
+            if not integration:
+                return MISSING_SERVICE
+
+            library = get_one(self._db, Library, short_name=library_short_name)
+            if not library:
+                return NO_SUCH_LIBRARY
+
+            auth_document_url = self.url_for("acquisition_groups", library_short_name=library.short_name)
```

Comment:
Ah, I misread the code. I thought this was the URL you GET the next line, but this is the URL you're POSTing as data on line 1938.